### DISCUSSION TITLE
International Pricing and Shipping Addresses

### DISCUSSION THREAD

**Discussion by sienatime** [Shopify/storefront-api-feedback#46] [Help] [answered]
Hey folks, I'm hoping to find some help with international pricing. I've been following this guide: https://shopify.dev/custom-storefronts/products/international-pricing#create-a-checkout and I'm seeing inconsistent behavior with the new "shipping address can change checkout currency" functionality. For example, if I begin a checkout in GBP but add a US shipping address, my checkout currency changes to USD. **I am only seeing this work with the default shop country (e.g. shop country is US, adding a US address to a non-USD checkout changes the currency to USD).** Any other combination retains the original checkout currency (e.g. shop country is US, checkout begins in USD, add a London address, checkout remains USD instead of changing to GBP, when GBP is also a supported currency). **Is this expected?** 

I am able to reproduce on the 2021-07 and 2021-10 Storefront APIs, and also on the web. I was also able to confirm that it is related to default shop **country**, and not currency - a shop whose default country is GB will flip to GBP but not to other supported currencies. I tried a shop with a default country of Australia and default currency of USD, and adding Australian addresses flips to AUD, but adding US addresses does not flip to USD.

**Comment by cbothner** [comment, marked as answer]
Hi @sienatime. Thanks for the question. Let me give a little more detail about the expected behavior, as well as a bug that we're working on a fix for.

Because price changes in checkout can be disruptive to customers and harmful to conversion, we update currency/pricing only when a merchant has implemented a specific [international pricing](https://help.shopify.com/en/manual/payments/shopify-payments/multi-currency/international-pricing) configuration. Specifically, we change the currency and price in checkout based on the shipping address _only_ when changing to or from a country that has a nonzero percentage adjustment or specific fixed prices. If the shipping address causes a change between two countries that both use base pricing, then we prioritize a smooth customer experience without a sudden currency change, since it's the same amount of money either way.

But there is a bug right now, which you've described. The home country is being mistaken for a customized country when it's being switched _to_. That means, if a customer ends up browsing in a context other than the shop's home country, but then enters a shipping address that's in the shop's home country, we're switching currency when we don't intend to.
